### PR TITLE
[#166816802] Use requireSecureChannels to disable unsecure channels

### PR DESCRIPTION
**Pull request by francescopersico**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#166816802](https://www.pivotaltracker.com/story/show/166816802): Aggiungere metadato al servizio che inibisce l'inoltro via mail dei messaggi inviati da un determinato servizio


New dependencies added: <a href='https://www.npmjs.com/package/applicationinsights'><code>applicationinsights</code></a> and <a href='https://www.npmjs.com/package/azure-storage'><code>azure-storage</code></a>.


<h2><a href="https://github.com/Microsoft/ApplicationInsights-node.js#readme">applicationinsights</a></h2>
<p>Author: Unknown</p>
<p>Description: Microsoft Application Insights module for Node.js</p>
<p>Homepage: <a href="https://github.com/Microsoft/ApplicationInsights-node.js#readme">https://github.com/Microsoft/ApplicationInsights-node.js#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>about 5 years ago</td></tr><tr><td>Last Updated</td><td>20 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>60</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/cls-hooked'><code>cls-hooked</code></a>, <a href='https://www.npmjs.com/package/continuation-local-storage'><code>continuation-local-storage</code></a>, <a href='https://www.npmjs.com/package/diagnostic-channel'><code>diagnostic-channel</code></a> and <a href='https://www.npmjs.com/package/diagnostic-channel-publishers'><code>diagnostic-channel-publishers</code></a></td></tr><tr><td>Keywords</td><td>exception monitoring, request monitoring, performance monitoring, application insights, microsoft, azure, cloud, tracing, telemetry, analytics and apm</td></tr>
</table>
This README is too long to show.



<h2><a href="http://github.com/Azure/azure-storage-node">azure-storage</a></h2>
<p>Author: Microsoft Corporation</p>
<p>Description: Microsoft Azure Storage Client Library for Node.js</p>
<p>Homepage: <a href="http://github.com/Azure/azure-storage-node">http://github.com/Azure/azure-storage-node</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 5 years ago</td></tr><tr><td>Last Updated</td><td>5 months ago</td></tr><tr><td>License</td><td>Apache-2.0</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>46</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/browserify-mime'><code>browserify-mime</code></a>, <a href='https://www.npmjs.com/package/extend'><code>extend</code></a>, <a href='https://www.npmjs.com/package/json-edm-parser'><code>json-edm-parser</code></a>, <a href='https://www.npmjs.com/package/md5.js'><code>md5.js</code></a>, <a href='https://www.npmjs.com/package/readable-stream'><code>readable-stream</code></a>, <a href='https://www.npmjs.com/package/request'><code>request</code></a>, <a href='https://www.npmjs.com/package/underscore'><code>underscore</code></a>, <a href='https://www.npmjs.com/package/uuid'><code>uuid</code></a>, <a href='https://www.npmjs.com/package/validator'><code>validator</code></a>, <a href='https://www.npmjs.com/package/xml2js'><code>xml2js</code></a> and <a href='https://www.npmjs.com/package/xmlbuilder'><code>xmlbuilder</code></a></td></tr><tr><td>Keywords</td><td>node, azure and storage</td></tr>
</table>
This README is too long to show.

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: CreateNotificationActivity/handler.ts [hunk 1]
```diff
@@ -170,10 +170,13 @@ export const getCreateNotificationActivityHandler = (
   //
   // * email notifications are enabled in the user profile (isEmailEnabledInProfile)
   // * email notifications are not blocked for the sender service (!isEmailBlockedForService)
+  // * the sender service do NOT require secure channels
```
Comment (cloudify) [inline CreateNotificationActivity/handler.ts]:
```suggestion
  // * the sender service does NOT require secure channels
```

File: CreateNotificationActivity/handler.ts [hunk 2]
```diff
@@ -170,10 +170,13 @@ export const getCreateNotificationActivityHandler = (
...
   // * a destination email address is configured (maybeEmailFromProfile)
   //
   const maybeEmailNotificationAddress =
-    isEmailEnabledInProfile && !isEmailBlockedForService
+    isEmailEnabledInProfile &&
+    !isEmailBlockedForService &&
+    !senderMetadata.requireSecureChannels
```
Comment (cloudify) [inline CreateNotificationActivity/handler.ts]: here I tried to keep the code at minimum to make it very easy to understand, I think it's better to create an intermediate variable in the block above like `const isEmailChannelAllowed = !senderMetadata.requireSecureChannels`

File: package.json [hunk 3]
```diff
@@ -39,13 +39,15 @@
     "typescript": "^3.3.3"
   },
   "dependencies": {
+    "applicationinsights": "^1.0.3",
```
Comment (cloudify) [inline package.json]: are these new versions needed for this PR? it doesn't seem so

Comment (francescopersico) [reply package.json]: It is in sync with the one in commons

File: CreateNotificationActivity/handler.ts (hunk 2 above)
Comment (francescopersico) [reply CreateNotificationActivity/handler.ts]: Done

**Pull request merged**